src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.ML
     /// </summary>
     public abstract class TrainCatalogBase
     {
-        protected internal readonly IHost Host;
+        protected internal IHostEnvironment Host { get; set; }
```

Comment:
>set [](start = 56, length = 3)

Why do you think you need a set?

Reply:
removed

---
In reply to: [256217066](https://github.com/dotnet/machinelearning/pull/2524#discussion_r256217066) [](ancestors = 256217066)